### PR TITLE
kitty: 0.27.1 -> 0.28.1

### PR DESCRIPTION
**Pull request by adamcstephens**
###### Description of changes

https://github.com/kovidgoyal/kitty/blob/v0.28.0/docs/changelog.rst#0280-2023-04-15

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/kitty/update.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+LATEST_TAG=$(curl ${GITHUB_TOKEN:+" -u \":$GITHUB_TOKEN\""} --silent https://api.github.com/repos/kovidgoyal/kitty/releases/latest | jq -r '.tag_name')
+LATEST_VERSION=$(echo ${LATEST_TAG} | sed 's/^v//')
+
+if [ "$OLD_VERSION" = "$LATEST_VERSION" ]; then
+  echo "kitty is already up-to-date at $OLD_VERSION"
+  exit 0
+fi
+
+HASH=$(nix-prefetch-url --quiet --unpack https://github.com/kovidgoyal/kitty/archive/refs/tags/${LATEST_TAG}.tar.gz)
+HASH=$(nix hash to-sri --type sha256 "$HASH")
```
Comment (winterqt) [inline pkgs/applications/terminal-emulators/kitty/update.sh:22]:
```suggestion
HASH=$(nix --experimental-features=nix-command hash to-sri --type sha256 "$HASH")
```

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/applications/terminal-emulators/kitty/update.sh (hunk 1 above)
Comment (adamcstephens) [reply pkgs/applications/terminal-emulators/kitty/update.sh]: FYI it's `--experimental-features nix-command`. The = is invalid apparently.

Comment (winterqt) [reply pkgs/applications/terminal-emulators/kitty/update.sh]: I forgot the CLI parser does not make any sense, thanks.

**Comment by Luflosi** [comment]
Could the update script also verify the GPG signature?

**Comment by adamcstephens** [comment]
> Could the update script also verify the GPG signature?

I've added verification of the git tag signature using the GitHub API.

**Review by amarshall — APPROVED** [review]
Verified builds and basic behavior works on x86_64-linux & x86_64-darwin. I haven’t reviewed the update script.

**Comment by adamcstephens** [comment]
I've removed the bash update script, as @lilyinstarlight pointed out that the latest nix-update could work with this python/go package. Split go-modules into its own attribute and verified the vendor hash is now updated by nix-update.

**Comment by adamcstephens** [comment]
@Luflosi do you have concerns about switching to nix-update instead of the bash script, given that it doesn't validate the GPG signature in any way? I'd appreciate your feedback/review on this if you have the time. Thanks!

**Comment by Luflosi** [comment]
I'm fine with using nix-update.
Since kitty hasn't been compatible with my old macOS version for some time now, I'm using an old version of kitty and am not particularly interested in maintaining the current version. I plan to actively maintain the kitty package again when I get a new laptop (with NixOS) some time this year.

File: pkgs/applications/terminal-emulators/kitty/default.nix [hunk 2]
```diff
@@ -99,16 +100,16 @@ buildPythonApplication rec {
   CGO_ENABLED = 0;
   GOFLAGS = "-trimpath";
 
-  configurePhase = let
-    goModules = (buildGoModule {
+  go-modules =(buildGoModule {
```
Comment (Luflosi) [inline pkgs/applications/terminal-emulators/kitty/default.nix:103]:
```suggestion
  go-modules = (buildGoModule {
```

**Comment by ocfox** [comment]
Result of `nixpkgs-review pr 226297` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>kitty</li>
    <li>kitty.kitten</li>
    <li>kitty.shell_integration</li>
    <li>kitty.terminfo</li>
  </ul>
</details>

**Review by ocfox — COMMENTED** [review]
Except for format error, LGTM, and works fine on darwin-aarch64.

File: pkgs/applications/terminal-emulators/kitty/default.nix (hunk 2 above)
Comment (adamcstephens) [reply pkgs/applications/terminal-emulators/kitty/default.nix]: Fixed. I look forward to a formatting standard...

**Review by lilyinstarlight — APPROVED** [review]
Looks good to me and seems to work correctly when testing on my system. Thank you!

**Pull request merged**